diffmatchpatch/diffmatchpatch_test.go

```diff
@@ -35,130 +39,134 @@ func pretty(diffs []Diff) string {
 		default:
 			_, _ = w.WriteString("Unknown")
 		}
+
 		_, _ = w.WriteString(fmt.Sprintf(": %v\n", diff.Text))
 	}
+
 	return w.String()
 }
 
-func assertDiffEqual(t *testing.T, seq1, seq2 []Diff) {
-	if a, b := len(seq1), len(seq2); a != b {
-		t.Errorf("%v\nseq1:\n%v\nseq2:\n%v", caller(), pretty(seq1), pretty(seq2))
-		t.Errorf("%v Sequences of different length: %v != %v", caller(), a, b)
-		return
-	}
+func diffRebuildTexts(diffs []Diff) []string {
+	texts := []string{"", ""}
 
-	for i := range seq1 {
-		if a, b := seq1[i], seq2[i]; a != b {
-			t.Errorf("%v\nseq1:\n%v\nseq2:\n%v", caller(), pretty(seq1), pretty(seq2))
-			t.Errorf("%v %v != %v", caller(), a, b)
-			return
+	for _, d := range diffs {
+		if d.Type != DiffInsert {
+			texts[0] += d.Text
 		}
-	}
-}
-
-func assertStrEqual(t *testing.T, seq1, seq2 []string) {
-	if a, b := len(seq1), len(seq2); a != b {
-		t.Fatalf("%v Sequences of different length: %v != %v", caller(), a, b)
-	}
-
-	for i := range seq1 {
-		if a, b := seq1[i], seq2[i]; a != b {
-			t.Fatalf("%v %v != %v", caller(), a, b)
+		if d.Type != DiffDelete {
+			texts[1] += d.Text
 		}
 	}
-}
 
-func diffRebuildtexts(diffs []Diff) []string {
-	text := []string{"", ""}
-	for _, myDiff := range diffs {
-		if myDiff.Type != DiffInsert {
-			text[0] += myDiff.Text
-		}
-		if myDiff.Type != DiffDelete {
-			text[1] += myDiff.Text
-		}
-	}
-	return text
+	return texts
 }
 
-func readFile(b *testing.B, filename string) string {
-	bytes, err := ioutil.ReadFile(filename)
+func readFile(filepath string) string {
+	data, err := ioutil.ReadFile(filepath)
 	if err != nil {
-		b.Fatal(err)
+		panic(err)
 	}
 
-	return string(bytes)
+	return string(data)
 }
 
-func speedtestTexts(b *testing.B) (s1 string, s2 string) {
-	s1 = readFile(b, "../testdata/speedtest1.txt")
-	s2 = readFile(b, "../testdata/speedtest2.txt")
+func speedtestTexts() (s1 string, s2 string) {
+	s1 = readFile("../testdata/speedtest1.txt")
+	s2 = readFile("../testdata/speedtest2.txt")
 
 	return s1, s2
 }
 
-func Test_diffCommonPrefix(t *testing.T) {
-	dmp := New()
-	// Detect any common suffix.
-	// Null case.
-	assert.Equal(t, 0, dmp.DiffCommonPrefix("abc", "xyz"), "'abc' and 'xyz' should not be equal")
+func TestDiffCommonPrefix(t *testing.T) {
+	type TestCase struct {
```

Comment:
I prefer to not reuse the TestCase types. The reason is that that way I can keep them inside the test function. They are not exported and the way a test function is made stays always the same without the need to find the right TestCase type.